### PR TITLE
Add Existing wc-api-dev Code

### PR DESCRIPTION
**Pull request by timmyc**
This is the initial commit of sorts which copies over existing code from `wc-api-dev`. I have added checks for the prior function names from `wc-api-dev` in hopes of not doubling up on hooks/filter logic while we migrate the hotfixes and such to this new plugin.

Ideally I'd like to keep changes here to a minimum, ship a 0.1.0, then circle back with additional PRs to update other items we have mentioned in the original issue.

For https://github.com/Automattic/wp-calypso/issues/18603

### CODE REVIEW

File: hotfixes/wc-calypso-bridge-mailchimp-no-redirect.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * add_settings_error is only available in UI contecext
```
Comment (justinshreve) [inline hotfixes/wc-calypso-bridge-mailchimp-no-redirect.php]: I think this brought over the wrong file: https://github.com/woocommerce/wc-api-dev/blob/master/hotfixes/wc-api-dev-mailchimp-no-redirect.php

**Comment by timmyc** [comment]
> Do we want to consider calling these something else while we have the chance?

Interesting to hear the background on the name! I would be fine with dropping all the hotfix files into the `inc` folder if that works for you.

I will bring over the MailChimp API code too.

**Comment by justinshreve** [comment]
>  I would be fine with dropping all the hotfix files into the inc folder if that works for you.

Yeah, let's do that.

**Comment by timmyc** [comment]
@justinshreve I moved all the hotfixes to `inc` and added the MailChimp API files as well. On my local vvv instance, with `MailChimp_Woocommerce` installed/activated - I can see all the endpoints as expected.

**Comment by justinshreve** [comment]
@timmyc I think that was the eventually plan for it. I know we also have some feedback about how they do a few things WP wise as well. Something we'll loop back to once it launches.

**Pull request merged**